docs/content/guides/dagster/branch-depl-with-snowflake.mdx

```diff
@@ -0,0 +1,348 @@
+    )
+
+
+@op(required_resource_keys={"snowflake"}, ins={"start": In(Nothing)})
+def clone_production_database(context):
+    context.resources.snowflake.execute_query(
+        f"CREATE DATABASE PRODUCTION_CLONE_{os.environ['DAGSTER_CLOUD_PULL_REQUEST_ID']} CLONE \"PRODUCTION\""
+    )
+
+
+@graph
+def clone_prod():
+    clone_production_database(start=drop_database_clone())
+```
+
+We’ve defined `drop_database_clone` and `clone_production_database` to utilize the [Snowflake resource](\_apidocs/libraries/dagster-snowflake#dagster_snowflake.snowflake_resource). The Snowflake resource will use the same configuration as the Snowflake I/O manager to generate a connection to Snowflake. However, while our I/O manager writes outputs to Snowflake, the Snowflake resource executes queries against Snowflake.
+
+We can then add the `clone_prod` job to our repository, configuring it with the resources corresponding to the current environment. We can modify the resource mapping by environment in our repository as follows:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/repository_v2.py startafter=start_resources endbefore=end_resources
+resource_defs = {
+    "branch": {
+        "snowflake_io_manager": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+        "snowflake": snowflake_resource.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+    },
+    "production": {
+        "snowflake_io_manager": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": "PRODUCTION",
+            }
+        ),
+        "snowflake": snowflake_resource.configured({**snowflake_config, "database": "PRODUCTION"}),
+    },
+}
+```
+
+Then, we can add the `clone_prod` job to our repository:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/repository_v2.py startafter=start_repository endbefore=end_repository
+@repository
+def repo():
+    ...
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[get_current_env()]),
+        clone_prod.to_job(resource_defs=resource_defs[get_current_env()]),
+    ]
+```
+
+In our `branch_deployments.yml` file, we’ll add an additional step to the existing `dagster_cloud_build_push` job that queues a run of `clone_prod` within each branch deployment upon launch. The `dagster_cloud_build_push` job is triggered on multiple pull request events: `opened`, `synchronize`, `reopen`, and `closed`. This means that upon future pushes to the branch, we'll trigger a run of `clone_prod`. The `if` condition below ensures that `clone_prod` will not be be run if the pull request is closed:
```

Comment:
I've made some changes to address 1, 2, and 3--hopefully the guide is clearer now and addresses these questions.

I'll need to double check with someone else about 4, and we can sync on 5 to figure out the issue here

Reply:
For 5, the issue was that it didn't immediately cross my mind to look at the job logs in dagit. Once I put the code to trigger the job in the GH action, i just assumed i would be able to see logs there. This might already exist in the guide, but a  mention that logs for the job execution can be found in your cloud branch deployment would be sufficient